metrics/src/main.py

```diff
@@ -18,31 +18,117 @@
 #   along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 import sys
+import asyncio
+import aiohttp
 import logging
 from time import sleep
+from datetime import datetime
 
-from logs import init_default_logger
-from collector import collect_metrics
-from config import MONITOR_INTERVAL, ERROR_TIMEOUT, NETWORK_NAME, PROXY_ENDPOINTS
+from src.logs import init_default_logger
+from src.collector import collect_metrics, download_metadata
+from src.config import (
+    NETWORK_NAME,
+    PROXY_ENDPOINTS,
+    METRICS_CHECK_INTERVAL,
+    METRICS_ERROR_CHECK_INTERVAL,
+    DB_CONNECTION_RETRIES,
+    DB_CONNECTION_INTERVAL,
+    OFFCHAIN_KEY,
+)
+from src.models import db
+from src.db import bootstrap_db
 
 logger = logging.getLogger(__name__)
 
 
+def run_migrations():
+    logger.info('Running database migrations...')
+    try:
+        from models import Address, TransactionCount
+
+        with db:
+            db.create_tables([Address, TransactionCount])
+        logger.info('Database migrations completed successfully.')
+    except Exception as e:
+        logger.error(f'Error running migrations: {e}')
+        sys.exit(1)
+
+
 def run_metrics_loop():
     if NETWORK_NAME not in PROXY_ENDPOINTS:
         logger.error(f'Unsupported network: {NETWORK_NAME}')
         sys.exit(1)
+
+    logger.info(f'Starting metrics collection loop for network: {NETWORK_NAME}')
+    last_run_date = None
+
     while True:
-        logger.info('Metrics collector iteration started...')
+        current_date = datetime.now().date()
+
+        if last_run_date is None or current_date > last_run_date:
```

Comment:
If you are running `collect_metrics` at the beginning of the day, then in `daily_transactions` field for this date in db will be in fact daily transactions count from the previous day. Is it supposed to work like this?

Reply:
yes